packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -29,7 +29,7 @@ var publicPath = '/';
 // `publicUrl` is just like `publicPath`, but we will provide it to our app
 // as %PUBLIC_URL% in `index.html` and `process.env.PUBLIC_URL` in JavaScript.
 // Omit trailing slash as %PUBLIC_PATH%/xyz looks better than %PUBLIC_PATH%xyz.
-var publicUrl = '';
+var publicUrl = paths.servedPath.slice(0, -1);
```

Comment:
Are you sure whether it requested just injecting ENV variable, or if it requested actually serving the app from `/prefix` in DEV?